comment/views.py

```diff
@@ -1,24 +1,37 @@
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from rest_framework import status
-from .models import Comment
+from .models import Comment, Post
 from .serializers import CommentSerializer
 
 # Create your views here.
 class CommentListView(APIView):
 
     def get(self, request): 
-        comments = Comment.objects.all()
+        try:
+            post_id = request.GET.get('post')
+        except:
+            return Response({"detail": "missing fields ['post']"}, status=status.HTTP_400_BAD_REQUEST)
+        if not Post.objects.filter(id=post_id).exists():
+            return Response({"detail": "Not found."}, status=status.HTTP_404_NOT_FOUND)
+        comments = Comment.objects.filter(post_id=post_id)
         serializer = CommentSerializer(comments, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
         
 
     def post(self, request):
-        author = request.data.get('author')
-        comment = request.data.get('comment')
-        if not author or not comment:
+        author = request.user
+        post_id = request.data.get('post')
+        content = request.data.get('content')
+        
+        if not author.is_authenticated:
+            return Response({"detail": "Authentication credentials not provided"}, status=status.HTTP_401_UNAUTHORIZED)
+        if not author or not content:
             return Response({"detail": "[author, comment] fields missing."}, status=status.HTTP_400_BAD_REQUEST)
-        comment = Comment.objects.create(author=author, comment=comment)
+        if not Post.objects.filter(id=post_id).exists():
+            return Response({"detail": "Not found."}, status=status.HTTP_404_NOT_FOUND)
+        post = Post.objects.get(id=post_id)
```

Comment:
```suggestion
        try:
            post = Post.objects.get(id=post_id)
        except:
            return Response({"detail": "Not found."}, status=status.HTTP_404_NOT_FOUND)
```
이렇게 적으면 Post table을 한 번만 뒤져도 되기 때문에 더 좋을 것 같습니당 😉